docs/DevGuide/Parallelization.md

```diff
@@ -275,18 +302,119 @@ during the next phase. Typically the `execute_next_phase` function should just
-the Charm++ manual. In the above Array example `array_proxy` is a `CProxy` and
-so all the documentation for Charm++ array proxies applies. SpECTRE always
-creates empty arrays with the constructor and requires users to insert however
-many elements they want and on which cores they want them to be placed. Note
-that load balancing calls may result in array elements being moved.
+
+There are some parallel components that are common to many executables.
+
+- Parallel::GlobalCache (a Parallel::Algorithms::Nodegroup)
+- Parallel::MutableGlobalCache (a Parallel::Algorithms::Group)
+- DgElementArray (a Parallel::Algorithms::Array)
+- observers::Observer (a Parallel::Algorithms::Group)
+- observers::ObserverWriter (a Parallel::Algorithms::Nodegroup)
+
+The MutableGlobalCache deserves special mention, which is why is has its own
+section with instructions on how to use it. See [Mutable items in the
+GlobalCache](#dev_guide_parallelization_mutable_global_cache).
+
+## 4. Placement {#dev_guide_parallelization_component_placement}
+
+The user has some control over where parallel components get placed on the
+resources it is running on. Here is a figure that illustrates how one may place
+parallel components.
+
+\image html charm_node_structure.png "Parallel component placement."
+
+In this example we are running on three (3) nodes that have four (4) cores each.
+For all our executables, we reserve one core of each node purely for
+communication purposes. Nothing else is run on this core. Because of this, what
+Charm++ calls a node, doesn't correspond to a full node on a supercomputer. A
+charm-node simply corresponds to a collection of cores that need not be on the
+same physical node. In our case, a charm-node is represented by the remaining
+cores on a node not used for communication (i.e. the first charm-node
+corresponds to cores 1-3 on the first physical node). Also the definition of a
+charm-core doesn't necessarily have to correspond to an actual core, however,
+for our purposes, it does.
+
+SpECTRE offers wrappers around Charm++ functions that will tell you the total
+number of charm-nodes/cores in an executable and what charm-node/core a parallel
+component is on. (In the following examples, the type `T` is usually an `int` or
+a `size_t`)
+
+- `Parallel::my_node<T>()` returns the charm-node that the parallel component is
+  on. In the figure, `Sing. 4` would return `2`.
+- `Parallel::my_proc<T>()` returns the charm-core that the parallel component is
+  on. In the figure, `Sing. 4` would return `6` (*not* `9`).
+- `Parallel::number_of_nodes<T>()` returns the total number of charm-nodes in an
+  executable. The above figure would have `3` charm-nodes.
+- `Parallel::number_of_procs<T>()` returns the total number of charm-cores in an
+  executable. The above figure would have `9` charm-cores (*not* `12`).
+
+\note For Charm++ SMP (shared memory parallelism) builds, a node corresponds to
```

Comment:
This is also not necessarily true. You can still have multiple Charm++ nodes per physical node. I don't remember exactly what you did on Anvil, but one way of having multiple comm threads per node is to have multiple virtual/Charm++ nodes per physical node